src/elements/frame_element.ts

```diff
@@ -13,97 +32,117 @@ export interface FrameElementDelegate {
   isLoading: boolean
 }
 
-export class FrameElement extends HTMLElement {
-  static delegateConstructor: new (element: FrameElement) => FrameElementDelegate
+export function frameElementFactory(Base: new() => HTMLElement) {
+  return class extends Base implements FrameElement {
+    readonly isTurboFrameElement: boolean = true
+    loaded: Promise<FetchResponse | void> = Promise.resolve()
+    readonly delegate: FrameElementDelegate
 
-  loaded: Promise<FetchResponse | void> = Promise.resolve()
-  readonly delegate: FrameElementDelegate
+    static get observedAttributes() {
+      return ["disabled", "loading", "src"]
+    }
 
-  static get observedAttributes() {
-    return ["disabled", "loading", "src"]
-  }
+    constructor() {
+      super()
+      if (!this.autonomous) {
+        this.setAttribute("is", this.isValue)
```

Comment:
That is intentional, and necessary when the element is programatically created, due to this part of that same section of the HTML standard: "when creating a customized built-in element programmatically, the _is_ attribute will not be present in the DOM, since it was not explicitly set". 

To implement its own behaviour, Turbo usually depends on the element name. Since that is no longer possible in this case, it must instead depend instead on the _is_ attribute being present in the DOM, to identify the element for update. Setting any other attribute would be a) DOM pollution, and b) a potential namespace clash. Using any other means would break the general design promise of progressively enhancing the HTML we already have, and amount to a redesign of Turbo's core behaviour which is definitely out of scope for this PR.

However I agree that it only needs setting if not already defined.

I hope to revisit this PR sometime this month, maybe with the revised polyfill it can even work on Safari, thanks!

Reply:
Using CE since 2014 and wrote all polyfills to date that work with built-in extends too, no need to explain me what's the `is` attribute about ;-) and yes, I do that in [vanilla-elements](https://github.com/WebReflection/vanilla-elements/blob/main/esm/main.js#L188-L192) too, but the reason I've mentioned that is that [the polyfill also does that](https://github.com/WebReflection/custom-elements-builtin/blob/master/esm/index.js#L81-L95) so you don't need to care.

Anyway, a simple check if the attribute is not defined yet is all I am pointing out, and shame on me I didn't check what was the `isValue` value, but if it's the custom element name, as builtin extend, then all good, this shouldn't interfere with the polyfill, I thought it was something else, like Vue 3 abuse it.

Apparently it's not 👍